src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -0,0 +1,959 @@
+            {
+                if (IsAsync)
+                {
+                    arguments.Diagnostics.Add(ErrorCode.ERR_SecurityCriticalOrSecuritySafeCriticalOnAsync, arguments.AttributeSyntaxOpt.Location, arguments.AttributeSyntaxOpt.GetErrorDisplayName());
+                }
+            }
+            else if (attribute.IsTargetAttribute(this, AttributeDescription.DoesNotReturnAttribute))
+            {
+                arguments.GetOrCreateData<MethodWellKnownAttributeData>().HasDoesNotReturnAttribute = true;
+            }
+            else
+            {
+                var compilation = this.DeclaringCompilation;
+                if (attribute.IsSecurityAttribute(compilation))
+                {
+                    attribute.DecodeSecurityAttribute<MethodWellKnownAttributeData>(this, compilation, ref arguments);
+                }
+            }
+        }
+
+        public override FlowAnalysisAnnotations FlowAnalysisAnnotations
+        {
+            get
+            {
+                return DecodeFlowAnalysisAttributes(GetDecodedWellKnownAttributeData());
+            }
+        }
+
+        private static FlowAnalysisAnnotations DecodeFlowAnalysisAttributes(MethodWellKnownAttributeData attributeData)
+            => attributeData?.HasDoesNotReturnAttribute == true ? FlowAnalysisAnnotations.DoesNotReturn : FlowAnalysisAnnotations.None;
+
+        private bool VerifyObsoleteAttributeAppliedToMethod(
+            ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments,
+            AttributeDescription description)
+        {
+            if (arguments.Attribute.IsTargetAttribute(this, description))
+            {
+                if (this.IsAccessor())
+                {
+                    if (this is SourceEventAccessorSymbol)
+                    {
+                        // CS1667: Attribute '{0}' is not valid on event accessors. It is only valid on '{1}' declarations.
+                        AttributeUsageInfo attributeUsage = arguments.Attribute.AttributeClass.GetAttributeUsageInfo();
+                        arguments.Diagnostics.Add(ErrorCode.ERR_AttributeNotOnEventAccessor, arguments.AttributeSyntaxOpt.Name.Location, description.FullName, attributeUsage.GetValidTargetsErrorArgument());
+                    }
+                    else
+                    {
+                        MessageID.IDS_FeatureObsoleteOnPropertyAccessor.CheckFeatureAvailability(arguments.Diagnostics, arguments.AttributeSyntaxOpt);
+                    }
+                }
+                else if (this.MethodKind is MethodKind.LocalFunction)
+                {
+                    // CS8760: Attribute '{0}' is not valid on local functions.
+                    arguments.Diagnostics.Add(ErrorCode.ERR_AttributeNotOnLocalFunction, arguments.AttributeSyntaxOpt.Name.Location, description.FullName);
+                }
```

Comment:
This doesn't agree with the draft spec in [https://github.com/dotnet/csharplang/issues/3008](https://github.com/dotnet/csharplang/issues/3008).  I don't know whether it is easier to make it work (it might just) or to update the spec.  I assume making it work is easier.